src/utils/__tests__/internal.ts

```diff
@@ -475,3 +478,41 @@ describe('createProcedureMethod', () => {
     sinon.assert.calledWithExactly(checkAuthorization, procArgs, context);
   });
 });
+
+describe('assertIsInteger', () => {
+  beforeAll(() => {
+    dsMockUtils.initMocks();
+  });
+
+  afterEach(() => {
+    dsMockUtils.reset();
+  });
+
+  afterAll(() => {
+    dsMockUtils.cleanup();
+  });
+
+  test('assertIsInteger should throw an error if the count is not an integer', async () => {
+    expect(() => assertIsInteger(('noInteger' as unknown) as BigNumber)).toThrow(
+      'The number must be an integer'
+    );
+  });
```

Comment:
You should be testing that the function throws when a float is passed. As we discussed, it's the compiler's job to validate that inputs are of the correct type